airflow/jobs/backfill_job.py

```diff
@@ -266,6 +275,11 @@ def _manage_executor_state(self, running):
                 )
                 self.log.error(msg)
                 ti.handle_failure_with_callback(error=msg)
+                continue
+            if ti.state not in self.STATES_COUNT_AS_RUNNING:
+                for node in ti.task.mapped_dependants():
+                    assert isinstance(node, MappedOperator)
```

Comment:
Is this guaranteed or just a temporary measure to get pass Mypy?

Reply:
Temporary for now until we support mapping task groups (the DAG api supports it, but the runtime doesn't yet)

Reply:
The why: Because `expand_mapped_task` doesn't exist on BaseOperator, only MappedOperator

Reply:
Makes sense